src/test/java/gov/usgs/wma/waterdata/ProcessTimeSeriesDescriptionIT.java

```diff
@@ -137,6 +138,33 @@ public void testUpdate() {
 	public void testNoUpdateIfSameData() {
 		// try to update data that is already current, no update, return no unique ids
 		ResultObject result = processTsd.processRequest(request);
-		assertEquals(Arrays.asList(), result.getTimeSeriesList());
+		assertEquals(Collections.emptyList(), result.getTimeSeriesList());
+	}
+
+	@Test
+	@DatabaseSetup("classpath:/testResult/empty/")
+	@ExpectedDatabase(
+			value = "classpath:/testResult/noDuplicateRowsForParmCode72019/",
+			assertionMode = DatabaseAssertionMode.NON_STRICT_UNORDERED)
+	public void testUpsertNotAffectingRowTwice() {
+		request.setId(TimeSeriesDescriptionDaoIT.JSON_DATA_ID_510);
+		ResultObject result = processTsd.processRequest(request);
+
+		assertThat(result.getTimeSeriesList(), containsInAnyOrder(
+				new TimeSeries("ff02b576d7d34f1c82a97e4405b16384"),
+				new TimeSeries("ed013f578a924df090ead63f05c5e5cc"),
+				new TimeSeries("739c687d86ee45308d09e9c8007f72b3"),
+				new TimeSeries("398ca443b25842a0a07d0e5741d25303"),
+				new TimeSeries("29be33c9a1ab4fa1aafb0ed97f2307ad"),
+				new TimeSeries("0f1ca4474f124e4ca3407e2aaa418c1c")
+				)
+		);
+
+		// This tests that the below error no longer occurs (was due to duplicate
+		// parm_cd in aq_comp_id_to_stat_cd table).
+		// ERROR: ON CONFLICT DO UPDATE command cannot affect row a second time
+		assertDoesNotThrow(() -> {
+			processTsd.processRequest(request);
+		}, "should not have thrown an exception but did");
```

Comment:
It look like my first comment got lost. 
I would say the assertDoesNotThrow() is not necessary.  Just call processTsd.processRequest(request);  if that
throws any exception, the test result should be flagged as ERROR.
assertThat() should be called again.